### PR TITLE
Update Godeps.json during tagging to point to tags

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/kubernetes/publishing-bot/issues/12

### CODE REVIEW

File: cmd/sync-tags/godeps.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		if err != nil {
+			return false, fmt.Errorf("failed to get refs/heads/%s in %q: %v", tag, p, err)
+		}
+		depRevisions[dep] = ref.Hash()
+	}
+
+	changed := false
+	for _, godep := range godeps.Deps {
+		for _, dep := range deps {
+			if godep.ImportPath == dep || !strings.HasPrefix(godep.ImportPath, dep+"/") {
```
Comment (caesarxuchao) [inline cmd/sync-tags/godeps.go]: Why is the negate?

File: cmd/sync-tags/main.go [hunk 2]
```diff
@@ -193,6 +206,34 @@ func main() {
 			continue
 		}
 
+		// update Godeps.json to point to actual tagged version in the dependencies. This version might differ
+		// from the one currently in Godeps.json because the other repo could have gotten more commit for this
+		// tag, but this repo didn't.
```
Comment (caesarxuchao) [inline cmd/sync-tags/main.go]: Add a link to https://github.com/kubernetes/publishing-bot/issues/12? The issue explains the problem with more details.

File: cmd/sync-tags/godeps.go (hunk 1 above)
Comment (sttts) [reply cmd/sync-tags/godeps.go]: typo

File: cmd/sync-tags/main.go (hunk 2 above)
Comment (sttts) [reply cmd/sync-tags/main.go]: 👍

**Comment by sttts** [comment]
Will do a test run with missing 1.9.0 tag (where we saw this problem).

**Pull request merged**